frontend/src/components/institute/NewSchoolPage.vue

```diff
@@ -0,0 +1,519 @@
+    </v-card-actions>
+  </v-card>
+</template>
+
+<script>
+import PrimaryButton from '../util/PrimaryButton';
+import {mapGetters, mapState} from 'vuex';
+import alertMixin from '@/mixins/alertMixin';
+import ApiService from '@/common/apiService';
+import {Routes} from '@/utils/constants';
+import * as Rules from '@/utils/institute/formRules';
+import {isNumber} from '@/utils/institute/formInput';
+
+export default {
+  name: 'NewSchoolPage',
+  mixins: [alertMixin],
+  props: {
+    districtNames: {
+      type: Array,
+      required: true
+    },
+    authorityNames: {
+      type: Array,
+      required: true
+    },
+  },
+  components: {
+    PrimaryButton,
+  },
+  mounted() {
+    this.validateForm();
+  },
+  data() {
+    return {
+      isFormValid: false,
+      processing: false,
+      newSchool: {
+        districtName: null,
+        authorityName: null,
+        openDate: null,
+        categoryCode: null,
+        facilityTypeCode: null,
+        organizationCode: null,
+        gradesOffered: null,
+        NLCActivity: null,
+        phoneNumber: null,
+        faxNumber: null,
+        email: null,
+        website: null,
+        mailingAddrLine1: null,
+        mailingAddrLine2: null,
+        mailingAddrCity: null,
+        mailingAddrProvince: null,
+        mailingAddrCountry: null,
+        mailingAddrPostal: null,
+        physicalAddrLine1: null,
+        physicalAddrLine2: null,
+        physicalAddrCity: null,
+        physicalAddrProvince: null,
+        physicalAddrCountry: null,
+        physicalAddrPostal: null,
+      },
+      rules: Rules,
+      newSchoolOpenDatePicker: null,
+      schoolFacilityTypes: [],
+      schoolCategoryTypes: [],
+      schoolOrganizationTypes: [],
+      schoolNeighborhoodLearningTypes: [],
+      schoolGradeTypes: [],
+      sameAsMailingCheckbox: true,
+      provinceCodeValues: [],
+      countryCodeValues: [],
+
+    };
+  },
+  computed: {
+    ...mapGetters('auth', ['isAuthenticated','userInfo']),
+    ...mapState('institute', ['facilityTypeCodes']),
+    ...mapState('institute', ['schoolCategoryTypeCodes']),
+    ...mapState('institute', ['schoolOrganizationTypeCodes']),
+    ...mapState('institute', ['schoolNeighborhoodLearningCodes']),
+    ...mapState('institute', ['gradeCodes']),
+    ...mapState('institute', ['provinceCodes']),
+    ...mapState('institute', ['countryCodes']),
+    ...mapState('institute', ['schoolCategoryFacilityTypesMap']),
+
+    allowedFacilityTypeCodesForSchoolCategoryCode(){
+      if(this.newSchool?.categoryCode){
+        return this.schoolCategoryFacilityTypesMap[this.newSchool?.categoryCode]?.map(schoolCatFacilityTypeCode =>  this.facilityTypeCodes.find(facTypCode=> facTypCode.facilityTypeCode === schoolCatFacilityTypeCode));
+      }else {
+        return [];
+      }
+    },
+  },
+  created() {
```

Comment:
created() method is the duplicate from SchoolDetails.vue. Not sure this in itself is an issue though.

Reply:
The SchoolDetails.vue allows the user to edit the school details so its form needs the same values in the dropdowns as this one. When the user creates a new school its from the school list screen so I don't see how this duplication can be avoided.